sound/soc/sof/intel/hda.c

```diff
@@ -495,28 +495,17 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 
 	/*
 	 * register our IRQ
-	 * let's try to enable msi firstly
-	 * if it fails, use legacy interrupt mode
-	 * TODO: support interrupt mode selection with kernel parameter
-	 *       support msi multiple vectors
+	 * looks msi has some interrupt missing issues for us ATM,
+	 * let's use legacy interrupt mode before that is fixed.
 	 */
-	ret = pci_alloc_irq_vectors(pci, 1, 1, PCI_IRQ_MSI);
-	if (ret < 0) {
-		dev_info(sdev->dev, "use legacy interrupt mode\n");
-		/*
-		 * in IO-APIC mode, hda->irq and ipc_irq are using the same
-		 * irq number of pci->irq
-		 */
-		hdev->irq = pci->irq;
-		sdev->ipc_irq = pci->irq;
-		sdev->msi_enabled = 0;
-	} else {
-		dev_info(sdev->dev, "use msi interrupt mode\n");
-		hdev->irq = pci_irq_vector(pci, 0);
-		/* ipc irq number is the same of hda irq */
-		sdev->ipc_irq = hdev->irq;
-		sdev->msi_enabled = 1;
-	}
+	dev_info(sdev->dev, "use legacy interrupt mode\n");
+	/*
+	 * use IO-APIC mode, hda->irq and ipc_irq are using the same
+	 * irq number of pci->irq
+	 */
+	hdev->irq = pci->irq;
+	sdev->ipc_irq = pci->irq;
+	sdev->msi_enabled = 0;
```

Comment:
it'd make more sense to have a kernel parameter for now that remove code that works fine on cAVS1.5.